### DISCUSSION TITLE
Supabase Storage Read Only Access policy for a particular folder

### DISCUSSION THREAD

**Discussion by ruhankhandakar** [supabase/supabase#18415] [Questions]
I have a private bucket (documents) and inside this buck I have couple of folders. By default all folders are accessible for authenticate users only. But I want public read access for one folder of documents bucket. What will be policy query for this? Please help me. Thanks

**Comment by GaryAustin1** [comment]
The example policies in the storage UI have something close to what you want:
```
CREATE POLICY "policy_name"
ON storage.objects FOR {operation} {USING | WITH CHECK} (
  -- restrict bucket
  bucket_id = {bucket_name}
  -- allow access to only jpg file
  AND storage."extension"(name) = 'jpg'
  -- in the public folder
  AND LOWER((storage.foldername(name))[1]) = 'public'
  -- to anonymous users
  AND auth.role() = 'anon'
);
```
Just drop the jpg test and change the folder name and bucket name.
You would only apply that for select policy.

**Reply by ruhankhandakar** [reply]
`((bucket_id = 'documents'::text) AND (storage.extension(name) = 'jpg'::text) AND (lower((storage.foldername(name))[2]) = 'profilePicture'::text) AND (auth.role() = 'anon'::text))`

set this policy
`https://jmrqrdddqxuadgbhkfrc.supabase.co/storage/v1/object/public/documents/profilePicture/8801728007477_profilePicture.jpg`
but still getting. 
`{"statusCode":"404","error":"Bucket not found","message":"Bucket not found"}` This error

**Reply by GaryAustin1** [reply]
You have public in your URL.  That is only used for the public URL to get an image from a public bucket.   You want /object/documents/profilePictore

**Reply by ruhankhandakar** [reply]
https://jmrqrdddqxuadgbhkfrc.supabase.co/storage/v1/object/documents/profilePicture/8801728007477_profilePicture.jpg

is this correct URL then ?
If not then how to access this file? which inside
documents(bucket name) -> profilePicture(folder name)
🙏🏻🙏🏻

**Reply by GaryAustin1** [reply]
That one looks correct.  
Are you providing an Authorization header with anon key it?

**Reply by ruhankhandakar** [reply]
> That one looks correct.  
> 
> Are you providing an Authorization header with anon key it?

No. I think it should open the image file directly in browser. But currently it's not happening.

Is this because of during bucket creation I haven't turn on "Public" option ?

**Reply by GaryAustin1** [reply]
You have to decide if you want the entire bucket public or not.  It can't be done on a folder basis.  If you make the bucket public then you can use just the publicURL to get the file.  If it is a private bucket you have to meet RLS to get the file.  With anon role check that means an Authorization header with 'bearer anon_key'.  I do not know if you remove that test it would work without an Authorization header or not for a private bucket.

**Reply by GaryAustin1** [reply]
Note that with a public bucket you DO have public in the path like you originally had.
